pytorch_lightning/trainer/trainer.py

```diff
@@ -1154,8 +1153,9 @@ def _run(
             self._call_callback_hooks("on_fit_start")
             self._call_lightning_module_hook("on_fit_start")
 
-        # plugin will move model to device
-        self._pre_dispatch()
+        # strategy will configure model and move it to the device
+        self.training_type_plugin.setup(self)
```

Comment:
Should the merged setup() happens before call `on_fit_start` hook.

Reply:
Yes we will have to figure out if it's possible or not. This branch is WIP and the PR is draft, let me explore all options before I make a proposal

Reply:
@awaelchli I know we discussed this in the past (but can't find the exact issue/PR) 

with the consolidation around spawning, is this the right spot at all for `on_fit_start`? what prevents this from being inside of the fit loop? 

is `fit` defined as: pretrain routine + sanity check + fit loop ? 

just for the sake of asking this: do we need `on_fit_start` / `on_fit_end` at all if we have the sanity check + on_train_start/end and on_validation_start/end hooks?

Reply:
Yes, exactly right. Although not a concern for this PR, the discussion of where to place on_fit_start can be restarted again. One can argue different ways:

- on_fit_start can move as early as possible inside the _run() method, but no earlier than setup_environment() since so far it was expected that the process group is initialized in this hook
- for consistency with `on_predict_start`, `on_test_start` etc., on_fit_start should move later into the fit_loop, but the pre-routine/sanity check hooks run earlier than that, making it a confusing choice
- call on_fit_start right when the pre_training_routine starts


>  do we need on_fit_start / on_fit_end at all

It seems like a natural choice to have it. We need to know what people do in this hook before we can decide what to do with it.

Reply:
Closing this conversation as it is diverging from the PR. Let's move this to an issue.